src/colorizer/loaders/JsonFeatureLoader.ts

```diff
@@ -15,9 +15,11 @@ export default class JsonFeatureLoader implements IFeatureLoader {
     const response = await fetch(url);
     const text = await response.text();
     const { data, min, max }: FeatureDataJson = JSON.parse(nanToNull(text));
+
     const tex = isBoolArray(data)
       ? packBooleanDataTexture(data)
       : packFloatDataTexture(data.map((val) => (val === null ? Infinity : val)));
-    return { tex, min, max };
+    const dataArr = isBoolArray(data) ? new Float32Array(data.map((value) => (value ? 1 : 0))) : new Float32Array(data);
+    return { data: dataArr, tex, min, max };
```

Comment:
Another spot where an abstract data source could be helpful. Besides being a bit awkward, the sanitization code for booleans is repeated from `packBooleanDataTexture`.